### PR TITLE
[SDK] Tests for new publish+deploy flow

### PR DESCRIPTION
**Pull request by kumaryash90**

### CODE REVIEW

File: packages/sdk/test/evm/new-publish-deploy.test.ts [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+      gatewayUrls: ["https://gateway.ipfscdn.io/ipfs/"],
+    });
+  });
+
+  describe("Deploy newly published contract", async () => {
+    let publishUriWithDirectDeploy: string;
+    let publishUriWithCustomFactoryDeploy: string;
+    let customFactoryPublishUri: string;
+
+    before("", async () => {
```
Comment (joaquim-verges) [inline packages/sdk/test/evm/new-publish-deploy.test.ts:29]: why is this in a before? can just be hardcoded above no?

File: packages/sdk/test/evm/new-publish-deploy.test.ts [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+        "ipfs://Qmc5cQotuEnCo71ZztJk69wc8kuKbGmuCARL2FyAojoi9r/";
+
+      // Split Main
+      customFactoryPublishUri =
+        "ipfs://QmZZUsE2hQbKc4pHPC3qjfXYrMvg8mceDqHxoQhgMudwau/";
+    });
+    it("should deploy regular contract", async () => {
+      const input = 27;
+      const contractAddress = await realSDK.deployer.deployContractFromUri(
+        publishUriWithDirectDeploy,
```
Comment (joaquim-verges) [inline packages/sdk/test/evm/new-publish-deploy.test.ts:45]: how about copying the data over in the test/asset folders and using mockStorage to pre-populate the storage/

File: packages/sdk/test/evm/new-publish-deploy.test.ts [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+      expect(number.toNumber()).to.equal(input);
+      expect(deployer).to.equal(adminWallet.address);
+    });
+
+    it("should deploy with custom factory", async () => {
+      // Re-upload with mock storage, enable all networks
+      const { compilerMetadata, extendedMetadata } =
+        await fetchAndCacheDeployMetadata(
+          customFactoryPublishUri,
+          realSDK.storage,
```
Comment (joaquim-verges) [inline packages/sdk/test/evm/new-publish-deploy.test.ts:62]: same here, you're downloading this data just to put it in mock storage, can't we just hardcode the data in our tests assets?

File: packages/sdk/test/evm/new-publish-deploy.test.ts [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+      const deployingAddress = await splitContract.call("splitMain");
+
+      expect(deployingAddress).to.equal(customFactoryAddress);
+    });
+  });
+
+  describe("Deploy re-published contract", async () => {
+    let publishUriWithAutoFactoryDeploy: string;
+
+    before("", async () => {
```
Comment (joaquim-verges) [inline packages/sdk/test/evm/new-publish-deploy.test.ts:146]: same here, don't need the before

File: packages/sdk/test/evm/new-publish-deploy.test.ts [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+      // Multiwrap
+      publishUriWithAutoFactoryDeploy =
+        "ipfs://QmQBgncSxFwLKusgAzXVpaWTuJTfV5yL6vs3jYXzz89EaY/";
+    });
+
+    it("should deploy with auto factory", async () => {
+      const mockPublisher = process.env.contractPublisherAddress;
+      process.env.contractPublisherAddress =
+        "0x664244560eBa21Bf82d7150C791bE1AbcD5B4cd7";
```
Comment (joaquim-verges) [inline packages/sdk/test/evm/new-publish-deploy.test.ts:155]: why did you need this?

File: packages/sdk/test/evm/new-publish-deploy.test.ts [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+
+      process.env.contractPublisherAddress = mockPublisher;
+    });
+  });
+
+  describe("Deploy old contract without re-publishing", async () => {
+    let publishUriWithDirectDeploy: string;
+    let publishUriWithFactoryDeploy: string;
+
+    before("", async () => {
```
Comment (joaquim-verges) [inline packages/sdk/test/evm/new-publish-deploy.test.ts:178]: same

File: packages/sdk/test/evm/new-publish-deploy.test.ts [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
+        );
+      await (
+        await thirdwebFactory.addImplementation(implementationAddress)
+      ).wait();
+
+      // Re-upload with mock storage, set implementation address for hardhat network
+      const { compilerMetadata, extendedMetadata } =
+        await fetchAndCacheDeployMetadata(
+          publishUriWithFactoryDeploy,
+          realSDK.storage,
```
Comment (joaquim-verges) [inline packages/sdk/test/evm/new-publish-deploy.test.ts:219]: same thing here, try to avoid IPFS altogether

File: packages/sdk/test/evm/new-publish-deploy.test.ts (hunk 5 above)
Comment (kumaryash90) [reply packages/sdk/test/evm/new-publish-deploy.test.ts:210]: when fetching a published contract for any-evm, it queries the polygon publisher
